src/pages/en/core-concepts/sharing-state.md

```diff
@@ -0,0 +1,685 @@
+
+Let's add a `cartItem` store to our `cartStore.js` from earlier. You can also switch to a TypeScript file to define the shape if you're so inclined 🙃
+
+
+<Tabs client:visible sharedStore="client-store-example">
+<Fragment slot="tab.js">clientStore.js</Fragment>
+<Fragment slot="tab.ts">clientStore.ts</Fragment>
+<Fragment slot="panel.js">
+```js
+import { atom, map } from 'nanostores';
+
+export const isCartOpen = atom(false);
+
+/**
+ * @typedef {Object} CartItem
+ * @property {string} id
+ * @property {string} name
+ * @property {string} imageSrc
+ * @property {number} quantity
+ */
+
+/** @type {import('nanostores').MapStore<Record<string, CartItem>>} */
+export const cartItems = map({});
+
+```
+</Fragment>
+<Fragment slot="panel.ts">
+```ts
+import { atom, map } from 'nanostores';
+
+export const isCartOpen = atom(false);
+
+export type CartItem = {
+  id: string;
+  name: string;
+  imageSrc: string;
+  quantity: number;
+}
+
+export const cartItems = map<Record<string, CartItem>>({});
+```
+</Fragment>
+</Tabs>
+
+Now, let's export an `addCartItem` helper for our components to use.
+- **If that item doesn't exist in your cart,** add the item with a starting quantity of 1.
+- **If that item _does_ already exist,** bump the quantity by 1.
+
+<Tabs client:visible sharedStore="client-store-example">
+<Fragment slot="tab.js">clientStore.js</Fragment>
+<Fragment slot="tab.ts">clientStore.ts</Fragment>
+<Fragment slot="panel.js">
+```js
+...
+export function addCartItem({ id, name, imageSrc }) {
+  const existingEntry = cartItems.get()[id];
+  if (existingEntry) {
+    cartItems.setKey(id, {
+      ...existingEntry,
+      quantity: existingEntry.quantity + 1,
+    })
+  } else {
+    cartItems.setKey(
+      id,
+      { id, name, imageSrc, quantity: 1 }
+    );
+  }
+}
+```
+</Fragment>
+<Fragment slot="panel.ts">
+```ts
+...
+type ItemDisplayInfo = Pick<CartItem, 'id' | 'name' | 'imageSrc'>;
+export function addCartItem({ id, name, imageSrc }: ItemDisplayInfo) {
+  const existingEntry = cartItems.get()[id];
+  if (existingEntry) {
+    cartItems.setKey(id, {
+      ...existingEntry,
+      quantity: existingEntry.quantity + 1,
+    });
+  } else {
+    cartItems.setKey(
+      id,
+      { id, name, imageSrc, quantity: 1 }
+    );
+  }
+}
+```
+</Fragment>
+</Tabs>
+
+:::note
+<details>
+
+<summary>**🙋 Why can use use `.get()` here instead of a `useStore` helper?**</summary>
+
+You may have noticed we're calling `cartItems.get()` here, instead of grabbing that `useStore` helper from our React / Preact / Solid / Vue examples. This is because **useStore is meant to trigger component re-renders.** In other words, `useStore` should be used whenever the store value is being rendered to the UI. Since and we're reading the value when an **event** is triggered (`addToCart` in this case), and we aren't trying to render that value, we don't need `useStore` here.
+</details>
```

Comment:
Just found these typos
```suggestion
<summary>**🙋 Why use `.get()` here instead of a `useStore` helper?**</summary>

You may have noticed we're calling `cartItems.get()` here, instead of grabbing that `useStore` helper from our React / Preact / Solid / Vue examples. This is because **useStore is meant to trigger component re-renders.** In other words, `useStore` should be used whenever the store value is being rendered to the UI. Since we're reading the value when an **event** is triggered (`addToCart` in this case), and we aren't trying to render that value, we don't need `useStore` here.
</details>
```